router/create.js

```diff
@@ -0,0 +1,51 @@
+import microCors from 'micro-cors';
+
+const isHappyMiddleware = (middlewareHandler) => middlewareHandler.length === 3; // (request, response, next)
+const isSadMiddleware = (middlewareHandler) => middlewareHandler.length === 4; // (error, request, response, next)
+
+function createRouter(...middlewareHandlers) {
+  const happyMiddlewares = middlewareHandlers.filter(isHappyMiddleware);
+  const sadMiddlewares = middlewareHandlers.filter(isSadMiddleware);
+
+  return function use(handler, options) {
+    const cors = microCors(options);
+
+    // eslint-disable-next-line require-yield
+    async function* routerGenerator(request, response, next) {
+      try {
+        // eslint-disable-next-line no-restricted-syntax
+        for await (const happyMiddleware of happyMiddlewares) {
+          yield await happyMiddleware(request, response, next);
+        }
```

Comment:
Se for assim já não resolve? Se resolver, podemos deixar uma função `async` simples :smile: 

```suggestion
        for (const happyMiddleware of happyMiddlewares) {
          await happyMiddleware(request, response, next);
        }
```